### ISSUE TITLE
[BUG] NNData has no Timestamp

### ISSUE THREAD

**Issue by justincdavis** [luxonis/depthai-core#913]
**Check if issue already exists**
Could not find any existing issues or forum posts regarding NNData timestamps

**Describe the bug**
The output of a neural network does not have a timestamp assigned to it. The NNData will always have a value of 0.0 for getTimestamp() and getTimestampDevice()

**Minimal Reproducible Example**
[MRE](https://drive.google.com/file/d/1Aj39ZMI9o0rkeJKMFydYX13ZfMvKOawY/view?usp=sharing)

**Screenshots**
![image](https://github.com/luxonis/depthai-python/assets/23462437/d33523f0-6c3e-4b62-94e0-1efaf4aa33fb)
PCL data age continues to increase since the timestamp is computed by taking the iteration timestamp - starting timestamp - data timestamp

**Expected behavior**
When calling getTimestamp or getTimestampDevice a non-zero timestamp would be returned with a value for when the NNData object was created.

**Attach system log**
{
    "architecture": "64bit WindowsPE",
    "machine": "AMD64",
    "platform": "Windows-10-10.0.22621-SP0",
    "processor": "AMD64 Family 25 Model 97 Stepping 2, AuthenticAMD",
    "python_build": "tags/v3.10.6:9c7b4bd Aug  1 2022 21:53:49",
    "python_compiler": "MSC v.1932 64 bit (AMD64)",
    "python_implementation": "CPython",
    "python_version": "3.10.6",
    "release": "10",
    "system": "Windows",
    "version": "10.0.22621",
    "win32_ver": "10 10.0.22621 SP0 Multiprocessor Free",
    "packages": [
        "absl-py==1.4.0",
        "Adafruit-Blinka==8.2.0",
        "adafruit-circuitpython-busdevice==5.2.3",
        "adafruit-circuitpython-motor==3.4.5",
        "adafruit-circuitpython-mpu6050==1.1.16",
        "adafruit-circuitpython-pca9685==3.4.6",
        "adafruit-circuitpython-register==1.9.13",
        "adafruit-circuitpython-requests==1.12.8",
        "adafruit-circuitpython-typing==1.8.1",
        "Adafruit-PlatformDetect==3.27.2",
        "Adafruit-PureIO==1.1.9",
        "addict==2.4.0",
        "aiogrpc==1.8",
        "aiohttp==3.8.4",
        "aiosignal==1.3.1",
        "alabaster==0.7.12",
        "astor==0.8.1",
        "astroid==2.12.12",
        "asttokens==2.2.1",
        "astunparse==1.6.3",
        "async-timeout==4.0.2",
        "attrs==22.2.0",
        "Babel==2.10.3",
        "backcall==0.2.0",
        "black==22.10.0",
        "bleach==5.0.1",
        "blobconverter==1.4.2",
        "board==1.0",
        "boto3==1.26.130",
        "botocore==1.29.130",
        "build==0.10.0",
        "bumpver==2023.1121",
        "CacheControl==0.12.11",
        "cachetools==5.3.0",
        "certifi==2022.9.24",
        "cffi==1.15.1",
        "chardet==5.1.0",
        "charset-normalizer==2.1.1",
        "cleo==2.0.1",
        "click==8.1.3",
        "colorama==0.4.6",
        "coloredlogs==15.0.1",
        "comm==0.1.3",
        "commonmark==0.9.1",
        "ConfigArgParse==1.5.3",
        "crashtest==0.4.1",
        "cvbridge3==1.1",
        "cycler==0.11.0",
        "dash==2.9.2",
        "dash-core-components==2.0.0",
        "dash-html-components==2.0.0",
        "dash-table==5.0.0",
        "debugpy==1.6.6",
        "decorator==5.1.1",
        "defusedxml==0.7.1",
        "depthai==2.22.0.0",
        "depthai-pipeline-graph==0.0.5",
        "depthai-sdk==1.11.0",
        "digi-xbee==1.4.1",
        "dill==0.3.6",
        "discord==2.1.0",
        "discord.py==2.2.3",
        "distlib==0.3.6",
        "docutils==0.18.1",
        "dulwich==0.21.5",
        "exceptiongroup==1.1.3",
        "executing==1.2.0",
        "fastjsonschema==2.15.3",
        "filelock==3.12.0",
        "flake8==5.0.4",
        "Flask==2.2.3",
        "flatbuffers==23.3.3",
        "flynt==1.0.1",
        "fonttools==4.37.1",
        "frozenlist==1.3.3",
        "future==0.18.3",
        "gast==0.4.0",
        "glfw==2.5.5",
        "google-auth==2.16.3",
        "google-auth-oauthlib==0.4.6",
        "google-pasta==0.2.0",
        "grpcio==1.51.3",
        "h5py==3.8.0",
        "html5lib==1.1",
        "humanfriendly==10.0",
        "idna==3.4",
        "imagesize==1.4.1",
        "imgui==1.4.1",
        "importlib-metadata==6.0.0",
        "iniconfig==2.0.0",
        "installer==0.7.0",
        "ipykernel==6.22.0",
        "ipython==8.12.0",
        "ipywidgets==8.0.6",
        "isort==5.10.1",
        "itsdangerous==2.1.2",
        "jaraco.classes==3.2.3",
        "jax==0.4.6",
        "jedi==0.18.2",
        "Jinja2==3.1.2",
        "jmespath==1.0.1",
        "joblib==1.1.0",
        "jsonschema==4.17.3",
        "jstyleson==0.0.2",
        "jupyter_client==8.1.0",
        "jupyter_core==5.3.0",
        "jupyterlab-widgets==3.0.7",
        "keras==2.12.0",
        "keyboard==0.13.5",
        "keyring==23.13.1",
        "kiwisolver==1.4.4",
        "kornia==0.6.12",
        "kornia_rs==0.0.8",
        "lazy-object-proxy==1.8.0",
        "lexid==2021.1006",
        "libclang==16.0.0",
        "llvmlite==0.38.1",
        "lockfile==0.12.2",
        "looseversion==1.1.2",
        "lxml==4.9.2",
        "Markdown==3.4.3",
        "markdown-it-py==2.2.0",
        "MarkupSafe==2.1.1",
        "marshmallow==3.17.0",
        "matplotlib==3.5.3",
        "matplotlib-inline==0.1.6",
        "mavsdk==1.4.1",
        "mccabe==0.7.0",
        "mdit-py-plugins==0.3.5",
        "mdurl==0.1.2",
        "more-itertools==9.0.0",
        "MouseInfo==0.1.3",
        "mpmath==1.3.0",
        "msgpack==1.0.5",
        "multidict==6.0.4",
        "mypy==0.982",
        "mypy-extensions==0.4.3",
        "myst-parser==1.0.0",
        "nbformat==5.7.0",
        "nest-asyncio==1.5.6",
        "networkx==2.8.8",
        "numba==0.55.1",
        "numpy==1.23.4",
        "oakutils @ file:///C:/Users/JustinDesktop/Code/oakutils",
        "oauthlib==3.2.2",
        "onnx==1.12.0",
        "onnxruntime==1.14.1",
        "onnxsim==0.4.28",
        "open3d==0.17.0",
        "opencv-contrib-python==4.8.0.76",
        "opencv-python==4.7.0.72",
        "openvino==2022.3.0",
        "openvino-dev==2022.3.0",
        "openvino-telemetry==2022.3.0",
        "openVO @ file:///C:/Users/JustinDesktop/Code/openVO",
        "opt-einsum==3.3.0",
        "packaging==21.3",
        "pandas==1.3.5",
        "parso==0.8.3",
        "pathlib2==2.3.7.post1",
        "pathspec==0.10.1",
        "pexpect==4.8.0",
        "pickleshare==0.7.5",
        "Pillow==9.2.0",
        "pip==23.2.1",
        "pkginfo==1.9.6",
        "platformdirs==3.5.1",
        "plotly==5.14.0",
        "pluggy==1.3.0",
        "poetry==1.5.0",
        "poetry-core==1.6.0",
        "poetry-plugin-export==1.3.1",
        "prompt-toolkit==3.0.38",
        "protobuf==3.20.1",
        "psutil==5.9.1",
        "pteranodon==0.0.1.1",
        "ptyprocess==0.7.0",
        "pure-eval==0.2.2",
        "pyasn1==0.4.8",
        "pyasn1-modules==0.2.8",
        "PyAutoGUI==0.9.53",
        "pybboxes==0.1.6",
        "pyclean==2.7.5",
        "pycodestyle==2.9.1",
        "pycparser==2.21",
        "pyflakes==2.5.0",
        "pyftdi==0.54.0",
        "PyGetWindow==0.0.9",
        "Pygments==2.16.1",
        "pyjion==1.2.6",
        "pylint==2.15.5",
        "pymavlink==2.4.37",
        "PyMsgBox==1.0.9",
        "PyNaCl==1.5.0",
        "PyOpenGL==3.1.6",
        "PyOpenGL-accelerate==3.1.6",
        "pyparsing==3.0.9",
        "pyperclip==1.8.2",
        "pyproject_hooks==1.0.0",
        "pyreadline3==3.4.1",
        "PyRect==0.2.0",
        "pyrsistent==0.19.3",
        "PyScreeze==0.1.28",
        "pyserial==3.5",
        "PySide2==5.15.2.1",
        "pytest==7.4.2",
        "python-dateutil==2.8.2",
        "python-dotenv==1.0.0",
        "pytube==15.0.0",
        "PyTurboJPEG==1.6.4",
        "pytweening==1.0.4",
        "pytz==2022.5",
        "pyupgrade==3.13.0",
        "pyusb==1.2.1",
        "pywin32==306",
        "pywin32-ctypes==0.2.0",
        "PyYAML==6.0",
        "pyzmq==25.0.2",
        "Qt.py==1.3.8",
        "rapidfuzz==2.15.1",
        "readme-renderer==37.3",
        "requests==2.29.0",
        "requests-oauthlib==1.3.1",
        "requests-toolbelt==0.10.1",
        "rfc3986==2.0.0",
        "rich==13.1.0",
        "rsa==4.9",
        "ruamel.yaml==0.17.21",
        "ruamel.yaml.clib==0.2.7",
        "ruff==0.0.287",
        "s3transfer==0.6.1",
        "scikit-learn==1.1.2",
        "scipy==1.11.2",
        "seaborn==0.12.2",
        "sentry-sdk==1.21.0",
        "setuptools==67.7.2",
        "shellingham==1.5.0.post1",
        "shiboken2==5.15.2.1",
        "simple-pid==1.0.1",
        "six==1.16.0",
        "sklearn==0.0",
        "smbus2==0.4.2",
        "snowballstemmer==2.2.0",
        "Sphinx==7.2.5",
        "sphinx-rtd-theme==1.3.0",
        "sphinxcontrib-applehelp==1.0.2",
        "sphinxcontrib-devhelp==1.0.2",
        "sphinxcontrib-htmlhelp==2.0.0",
        "sphinxcontrib-jquery==4.1",
        "sphinxcontrib-jsmath==1.0.1",
        "sphinxcontrib-qthelp==1.0.3",
        "sphinxcontrib-serializinghtml==1.1.9",
        "stack-data==0.6.2",
        "sympy==1.11.1",
        "tenacity==8.2.2",
        "tensorboard==2.12.0",
        "tensorboard-data-server==0.7.0",
        "tensorboard-plugin-wit==1.8.1",
        "tensorflow==2.12.0",
        "tensorflow-estimator==2.12.0",
        "tensorflow-intel==2.12.0",
        "tensorflow-io-gcs-filesystem==0.31.0",
        "termcolor==2.2.0",
        "texttable==1.6.7",
        "threadpoolctl==3.1.0",
        "tokenize-rt==5.2.0",
        "toml==0.10.2",
        "tomli==2.0.1",
        "tomlkit==0.11.6",
        "torch==1.13.0",
        "torchaudio==0.13.0+cu117",
        "torchvision==0.14.0",
        "tornado==6.2",
        "tqdm==4.65.0",
        "traitlets==5.9.0",
        "trove-classifiers==2023.5.2",
        "twine==4.0.2",
        "types-PySide2==5.15.2.1.5",
        "typing_extensions==4.3.0",
        "urllib3==1.26.12",
        "vext==0.7.6",
        "vext.gi==0.7.4",
        "virtualenv==20.23.0",
        "wcwidth==0.2.6",
        "webencodings==0.5.1",
        "websockets==3.4",
        "Werkzeug==2.2.3",
        "wheel==0.37.1",
        "widgetsnbextension==4.0.7",
        "wrapt==1.14.1",
        "xmltodict==0.13.0",
        "yacs==0.1.8",
        "yarl==1.8.2",
        "youtube-dl @ git+https://github.com/ytdl-org/youtube-dl.git@211cbfd5d46025a8e4d8f9f3d424aaada4698974",
        "zipp==3.11.0"
    ],
    "usb": [
        "No USB backend found"
    ],
    "uname": [
        "Windows DESKTOP-JUSTIN 10 10.0.22621 AMD64 AMD64 Family 25 Model 97 Stepping 2, AuthenticAMD"
    ]
}
